### ISSUE TITLE
GUI freezes unrecoverable when loading tracks

### ISSUE THREAD

**Issue by Holzhaus** [Holzhaus/mixxx-gh-issue-migration#267]
This is a follow-up from https://bugs.launchpad.net/mixxx/+bug/1789059/comments/16 and onwards. It is different from https://bugs.launchpad.net/mixxx/+bug/1743266 in that it is reproducible with waveform types other than RGB (GLSL), the tracks keep playing until the end, and I can forcefully close Mixxx without losing control of my desktop environment.

Description: Mixxx freezes when loading tracks and requires a force-close. Sometimes I can load 50+ tracks before it happens, other times I get it when first loading a track having just started Mixxx. Regardless, it's been happenning consistently. Using a kernel with the realtime patchset and low-latency ('CONFIG_HZ=1000') appears to have made this more noticeable, but is not a requirement for reproducibility.

I was asked to test with a build without perftools from Google but the problem remains.

Relevant backtrace is at #23 in the linked bug; I'm attaching it again here for convenience. From it, developer uklotzde has identified a deadlock to be originating from 'WOverview'.

------------------------------------
Imported from Launchpad using lp2gh.

 * date created: 2020-07-01T10:29:47Z
 * owner: marc.2377
 * the launchpad url was https://bugs.launchpad.net/bugs/1885894

**Comment by Holzhaus** [comment]
(by marc.2377)

**Comment by Holzhaus** [comment]
(by be.ing)
What version of Qt are you using? The backtrace looks like the deadlock is in Qt.

**Comment by Holzhaus** [comment]
(by marc.2377)
It's QT 5.15.0.

**Comment by Holzhaus** [comment]
(by be.ing)
https://bugreports.qt.io/browse/QTBUG-84857

A comment on the upstream bug says it should be fixed in Qt 5.15.1. Could you try building Qt's 5.15 branch from their Git repository? https://github.com/mixxxdj/mixxx/wiki/compiling on linux has instructions for building Mixxx with a locally built version of Qt.

**Comment by Holzhaus** [comment]
(by be.ing)
Ugh Launchpad is awful and broke the link. Hopefully Launchpad doesn't break this link:
https://github.com/mixxxdj/mixxx/wiki/compiling-on-linux

**Comment by Holzhaus** [comment]
(by marc.2377)
Funny thing is, I actually tried to land on that page via Google (first result to https://www.google.com/search?client=firefox-b-d&q=mixxxx+switch+to+cmake) and it's broken there too. Had to refer to the cached version :p

Thanks for the finding @Be. I'll try it.

**Comment by Holzhaus** [comment]
(by uklotzde)
Should we set the state of this bug to "Won't fix"? We just have to remember NOT to use Qt 5.15.0 for any build.

Once again Fedora's strategy to adopt new Qt major versions only after 1 or 2 minor versions turns out to be a sensible decision.

**Comment by Holzhaus** [comment]
(by marc.2377)
Alright, compiling all QT5 packages at branch 5.15 proved to be a lengthy, slighly error-prone and painful endeavour, and when qtlocation would not build due to whatever error in the source code, I decided to try something quicker and apply the patch to https://bugreports.qt.io/browse/QTBUG-84619 on top of 5.15.0 instead. And as expected, the problem seems to have been fixed.

Thank you for your guidance and help!
